hibernate-core/src/main/java/org/hibernate/metamodel/mapping/internal/MappingModelCreationProcess.java

```diff
@@ -130,20 +134,27 @@ private void executePostInitCallbacks() {
 				}
 			}
 
-			if ( ! anyCompleted ) {
+			if ( !anyCompleted ) {
 				// none of the remaining callbacks could complete fully, this is an error
-				final StringBuilder buff = new StringBuilder(
-						"MappingModelCreationProcess$PostInitCallback queue could not be processed..."
-				);
-
-				postInitCallbacks.forEach(
-						callbackEntry -> buff.append( EOL )
-								.append( "        - " ).append( callbackEntry )
-				);
-
-				buff.append( EOL );
-
-				throw new IllegalStateException( buff.toString() );
+				if ( !exceptions.isEmpty() ) {
+					throw new IllegalStateException(
+							"MappingModelCreationProcess$PostInitCallback queue could not be processed...",
+							exceptions.entrySet().iterator().next().getValue()
+					);
```

Comment:
I would prefer if we add all exceptions as suppressed exceptions

Reply:
Or create a composite exception as http://reactivex.io/RxJava/3.x/javadoc/io/reactivex/rxjava3/exceptions/CompositeException.html
We might well create a common class to be reused elsewhere.